contributors/devel/sig-architecture/api-conventions.md

```diff
@@ -308,17 +308,49 @@ response reduces the complexity of these clients.
 
 ##### Typical status properties
 
-**Conditions** represent the latest available observations of an object's
-state.  They are an extension mechanism intended to be used when the details of
-an observation are not a priori known or would not apply to all instances of a
-given Kind.  For observations that are well known and apply to all instances, a
-regular field is preferred.  An example of a Condition that probably should
-have been a regular field is Pod's "Ready" condition - it is managed by core
-controllers, it is well understood, and it applies to all Pods.
+**Conditions** provide a standard mechanism for higher-level status reporting
+from a controller. They are an extension mechanism which allows tools and other
+controllers to collect summary information about resources without needing to
+understand resource-specific status details. Conditions should complement more
+detailed information about the observed status of an object written by a
+controller, rather than replace it. For example, the "Available" condition of a
+Deployment can be determined by examining `readyReplicas`, `replicas`, and
+other properties of the Deployment. However, the "Available" condition allows
+other components to avoid duplicating the availability logic in the Deployment
+controller.
 
 Objects may report multiple conditions, and new types of conditions may be
 added in the future or by 3rd party controllers. Therefore, conditions are
-represented using a list/slice, where all have similar structure.
+represented using a list/slice of objects, where each condition has a similar
+structure. Conditions are most useful when they follow some consistent
+conventions:
+
+* Conditions should be added to explicitly convey properties that users and
+  components care about rather than requiring those properties to be inferred
+  from other observations.  Once defined, the meaning of a Condition can not be
```

Comment:
Maybe worth expanding on "meaning"?  Specifically roll-up sorts of signals are valuable because consumers do not have to do look at N raw signals and make the same inference AND because that set of N raw signals can evolve into M raw signals, and the roll-up is handled automatically.

See for example pod readiness gates - the raw signals got more complex, but consumers only had to worry about the "Ready" condition.

In that case the internal composition of "Ready" changed, but the external semantic did not.

Reply:
We can't really enforce this in code, so users can't really rely on this. I'd therefore phrase it as a recommendation rather than requirement.

Reply:
I think "meaning" here is the "intent" of the Condition -- it may become possible to more accurately measure the intent.

For example, adding "is there enough disk space" to the set of conditions that indicate "this Node is able to accept work" or "the Pods on this Node are likely to function properly" -- but it would not be acceptable to swap "able to accept work" intent for an "Pods function properly" intent; the latter would need to be a new Condition.

Reply:
My original comment was that the text could be clearer about what "meaning" means - how abstract that is vs how it is literally derived.

But I think it also unclear if you mean that conditions should exclusively be used as "summaries" of other information or that "summaries" are one of several reasonable uses of conditions?

And I am still looking for answers to my question on principle.  Why is a "ready" pronouncement *better* a a condition than as a field?

Reply:
My read is that conditions have consistent metadata attached to them (at least as of #1624) as to when and why they last transitioned states, including now the `observedGeneration`, which was previously not specific to conditions, so I guess that could be considered a new advantage of conditions over fields, or alternatively a new disadvantage of fields as it may make non-condition metadata harder to reason about. I suppose it doesn't stop anyone from defining `status.observedGeneration` as well, but overall having both seems like potential API bloat.